kernel/include/nano_internal.h

```diff
@@ -82,6 +82,56 @@ static inline unsigned int _Swap(unsigned int key)
 
 #ifdef CONFIG_USERSPACE
 /**
+ * @brief get the maximum number of free regions for memory domain partitions
+ *
+ * @return Max number of free regions
+ */
+extern u8_t _arch_get_max_partitions(void);
+
+/**
+ * @brief Configure MPU memory domain
+ *
+ * This function configures per thread memory domain reprogramming the MPU.
+ * Used when adding the currently running thread to a domain.
```

Comment:
This needs clarity: what does "configure" mean when the function doesn't actually take any configuration parameters?  Maybe call this "init" or "enable"?

Reply:
It essentially means "apply the configuration to the hardware". Agree this could be explained better.